src/components/wizard/wizard.component.ts

```diff
@@ -146,33 +152,31 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
     }
 
     private _step: number = 0;
-    private _onDestroy = new Subject<void>();
+    protected readonly _onDestroy = new Subject<void>();
 
-    ngAfterViewInit(): void {
+    constructor(protected readonly _wizardService: WizardService<WizardStepComponent>) {
+    }
 
+    ngOnInit(): void {
         // initially set the correct visibility of the steps
         setTimeout(this.update.bind(this));
 
-        // initially set the ids for each step
-        this.setWizardStepIds();
-
         // if the steps change then update the ids
-        this.steps.changes.pipe(tick(), takeUntil(this._onDestroy)).subscribe(() => {
-            this.setWizardStepIds();
-            this.update();
-        });
+        this.steps.changes.pipe(tick(), takeUntil(this._onDestroy)).subscribe(() => this.update());
+
+        // watch for changes to valid subject
+        this._wizardService.valid$.pipe(
+            filter((event: WizardValidEvent<WizardStepComponent>) => !event.valid),
+            takeUntil(this._onDestroy)
+        ).subscribe(() => this.setNextStepsUnvisited());
```

Comment:
This should be setting steps unvisited from the invalid step onwards, not the active step. The `valid` property can be changed on any step via the binding.